ziti/cmd/ascode/download/download.go

```diff
@@ -0,0 +1,407 @@
+/*
+	Copyright NetFoundry Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package download
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/judedaryl/go-arrayutils"
+	"github.com/michaelquigley/pfxlog"
+	"github.com/openziti/edge-api/rest_management_api_client"
+	"github.com/openziti/ziti/internal"
+	"github.com/openziti/ziti/internal/rest/mgmt"
+	"github.com/openziti/ziti/ziti/cmd/edge"
+	c "github.com/openziti/ziti/ziti/constants"
+	"github.com/patrickmn/go-cache"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"io"
+	"os"
+	"slices"
+	"strings"
+)
+
+var log = pfxlog.Logger()
+
+type Download struct {
+	loginOpts edge.LoginOptions
+	client    *rest_management_api_client.ZitiEdgeManagement
+
+	verbose bool
+
+	ofJson bool
+	ofYaml bool
+
+	file     *os.File
+	filename string
+
+	configCache      *cache.Cache
+	configTypeCache  *cache.Cache
+	authPolicyCache  *cache.Cache
+	externalJwtCache *cache.Cache
+
+	Out io.Writer
+	Err io.Writer
+}
+
+var output Output
+
+func NewDownload(loginOpts edge.LoginOptions, client *rest_management_api_client.ZitiEdgeManagement, ofJson bool, ofYaml bool, file *os.File, filename string) Download {
+	d := Download{}
+	d.loginOpts = loginOpts
+	d.client = client
+	d.ofJson = ofJson
+	d.ofYaml = ofYaml
+	d.file = file
+	d.filename = filename
+	return d
+}
+
+func NewDownloadCmd(out io.Writer, errOut io.Writer) *cobra.Command {
+
+	d := &Download{}
+	d.Out = out
+	d.Err = errOut
+	d.loginOpts = edge.LoginOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "export [entity]",
+		Short: "Export Ziti entities",
```

Comment:
mind removing the " ziti " parts of these descriptions? We know it's a ziti entity :)